### PR TITLE
First draft of changelog

### PR DESCRIPTION
**Pull request by katowulf**
@jwngr

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+important - We have a migration guide here: https://github.com/firebase/angularfire/releases/tag/v1.0.0
+removed - `$firebase` has been removed; see migration guide for alternatives
+removed - `$inst() was removed; use `$ref()` to get to the Firebase ref
+changed - `$asArray()` has been replaced by calling the `$firebaseArray` service directly
+changed - `$asObject()` has been replaced by calling the `$firebaseObject` service directly
+changed - `$FirebaseArray` renamed to $firebaseArray`
+changed - `$FirebaseObject` renamed to `$firebaseObject`
+changed - `$extendFactory` has been renamed to `$extend` (e.g. `$firebaseObject.$extend(...)`)
+feature - enhanced performance of $bindTo (thanks @jamesstalmage!)
+feature - improved test unit coverage, particularly around auth (thanks @jamesstalmage!)
```
Comment (jwngr) [inline changelog.txt]: Not a customer-facing change, so should not be in changelog.

File: changelog.txt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+added - An error message is now displayed if one tries to sync to array-like data, since this is error-prone
```
Comment (jwngr) [inline changelog.txt]: It's a warning, not an error message.

File: changelog.txt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+fixed - Travis now runs e2e locally with Firefox (thanks @jamesstalmage!)
```
Comment (jwngr) [inline changelog.txt]: Not a customer-facing change, so should not be in changelog.

File: changelog.txt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+fixed - $value is now removed when data changes from a primitive to an object with child keys
+fixed - better test consistency with Jasmine/Travis, less timing issues (thanks @jamesstalmage!)
```
Comment (jwngr) [inline changelog.txt]: Not a customer-facing change, so should not be in changelog.

File: changelog.txt [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+fixed - tests now work around new Jasmine version's breaking change to equality checks
```
Comment (jwngr) [inline changelog.txt]: Not a customer-facing change, so should not be in changelog.

File: changelog.txt [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+important - API is now official! Future 1.x versions will remain backwards compatible with the current API contract.
+important - We have a migration guide here: https://github.com/firebase/angularfire/releases/tag/v1.0.0
+removed - `$firebase` has been removed; see migration guide for alternatives
+removed - `$inst() was removed; use `$ref()` to get to the Firebase ref
+changed - `$asArray()` has been replaced by calling the `$firebaseArray` service directly
+changed - `$asObject()` has been replaced by calling the `$firebaseObject` service directly
+changed - `$FirebaseArray` renamed to $firebaseArray`
+changed - `$FirebaseObject` renamed to `$firebaseObject`
+changed - `$extendFactory` has been renamed to `$extend` (e.g. `$firebaseObject.$extend(...)`)
+feature - enhanced performance of $bindTo (thanks @jamesstalmage!)
```
Comment (jwngr) [inline changelog.txt]: All sentences should start with a capital letter and end with a period.

**Comment by jwngr** [comment]
I took a shot at updating the changelog and adding a few more changes.

**Comment by katowulf** [comment]
Looks good. Anything left for me to address?

**Comment by jwngr** [comment]
Let's hold off on merging this in just yet. Thanks for taking a look though!

**Pull request merged**